Toggl.Foundation.MvvmCross/Transformations/TimeEntriesLogTransformation.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Toggl.Foundation.Extensions;
+using Toggl.Foundation.Models.Interfaces;
+using Toggl.Foundation.MvvmCross.Collections;
+using Toggl.Foundation.MvvmCross.ViewModels;
+using Toggl.Foundation.MvvmCross.ViewModels.TimeEntriesLog;
+using Toggl.Multivac;
+using Toggl.Multivac.Extensions;
+
+namespace Toggl.Foundation.MvvmCross.Transformations
+{
+    internal static class TimeEntriesLogTransformation
+    {
+        public static IImmutableList<CollectionSection<DaySummaryViewModel, CollectionSection<TimeEntriesGroupViewModel, TimeEntryViewModel>>> Group(
+            IEnumerable<IThreadSafeTimeEntry> timeEntries, IThreadSafePreferences preferences, DateTimeOffset currentDateTime)
+            => timeEntries
+                .Where(isNotRunning)
+                .Select(te => new TimeEntryViewModel(te, preferences.DurationFormat))
+                .OrderByDescending(te => te.StartTime)
+                .GroupBy(te => te.StartTime.LocalDateTime.Date)
+                .Select(group => transform(group, preferences, currentDateTime))
+                .ToImmutableList();
+
+        private static bool isNotRunning(IThreadSafeTimeEntry timeEntry) => !timeEntry.IsRunning();
+
+        private static CollectionSection<DaySummaryViewModel, CollectionSection<TimeEntriesGroupViewModel, TimeEntryViewModel>> transform(
+            IGrouping<DateTime, TimeEntryViewModel> dayGroup, IThreadSafePreferences preferences, DateTimeOffset currentDateTime)
+            => new CollectionSection<DaySummaryViewModel, CollectionSection<TimeEntriesGroupViewModel, TimeEntryViewModel>>(
+                new DaySummaryViewModel(
+                    DateToTitleString.Convert(dayGroup.Key, currentDateTime),
+                    totalTrackedTime(dayGroup).ToFormattedString(preferences.DurationFormat)),
+                toGroups(dayGroup, preferences));
+
+        private static TimeSpan totalTrackedTime(IEnumerable<TimeEntryViewModel> timeEntries)
+            => timeEntries.Sum(timeEntry => timeEntry.Duration ?? TimeSpan.Zero);
```

Comment:
There's an ext. method to simplify this:

```suggestion
            => timeEntries.Sum(timeEntry => timeEntry.Duration);
```